bundles/org.openhab.core.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigUtil.java

```diff
@@ -35,14 +40,89 @@
  * except if a {@link ConfigDescriptionParameter} is given. In the latter case,
  * a conversion according to the type given in the description is performed.
  *
- * @author Kai Kreuzer - Initial API and implementation
+ * @author Kai Kreuzer - Initial contribution
  * @author Thomas Höfer - Minor changes for type normalization based on config description
  */
 public class ConfigUtil {
+
+    private static final String DEFAULT_LIST_DELIMITER = ",";
+
+    /**
+     * Maps the provided (default) value of the given {@link Type} to the corresponding Java type.
+     *
+     * In case the provided value is supposed to be a number and cannot be converted into the target type correctly,
+     * this method will return <code>null</code> while logging a warning.
+     *
+     * @param parameterType the {@link Type} of the value (must not be null)
+     * @param defaultValue the value that should be converted (must not be null)
+     * @return the given value as the corresponding Java type or <code>null</code> if the value could not be converted
+     */
+    public static @Nullable Object normalizeDefaultType(Type parameterType, String defaultValue) {
+        try {
+            switch (parameterType) {
+                case TEXT:
+                    return defaultValue;
+                case BOOLEAN:
+                    return Boolean.parseBoolean(defaultValue);
+                case INTEGER:
+                    return new BigInteger(defaultValue);
```

Comment:
This is a major change as it returned a `BigDecimal` before which is imo wrong.

Reply:
Not really - similar to the item types, we use `BigDecimal` for any number (be it int or float), so that consuming code can always cast to `BigDecimal` without having to worry.
The reason that we introduced a separation between INTEGER and DECIMAL for parameters was due to the JSON deserialisation - afair, integers always ended up being floats in Java (or something like that...).

Reply:
Good point.

We than have to be optimistic again because a decimal will be a valid default value for a integer config parameter. Should we maybe add a rounding strategy?

Reply:
Yes - maybe the same here: Log a warning and round it (down), so that no subsequent exceptions occur.

Reply:
I changed it back to `BigDecimal`, added a warning and round down.

btw. do we have a proper solution to verify logger outputs in our unit tests?

Reply:
I am not aware of any such solution...